core/txpool/txpool.go

```diff
@@ -813,6 +813,15 @@ func (pool *TxPool) add(tx *types.Transaction, local bool) (replaced bool, err e
 func (pool *TxPool) isFuture(from common.Address, tx *types.Transaction) bool {
 	list := pool.pending[from]
 	if list == nil {
+		// Transactions are first inserted into the queued list and bubbled up from there.
+		// Thus we also need to check the queued list as well.
+		queued := pool.queue[from]
+		if queued != nil {
+			if old := queued.txs.Get(tx.Nonce()); old != nil {
+				return false
+			}
+			return queued.txs.Get(tx.Nonce()-1) != nil
```

Comment:
This is wrong. 
First, didn't you mean to return `false` if the previous  nonce was present? 

Secondly, it would still be wrong
```
account.nonce = 10
queued has : [ tx with nonce 200]
```
And we get `tx with nonce 201` then `IsFuture ->  false`. 

TO do it right, we need to check if the transactions in queued constitues a gap-free sequence from `tx.nonce` down to `state.nonce` . Not sure it's worth it, personally.. (?)

Reply:
Yep I know, I discussed that with @rjl493456442 on discord today

Reply:
It's not correct. 

It can happen that account has no pending transaction, but has a few transactions in queue but they are executable(nonce is continuous, but just waiting for promotion) and the new one is continuous with txs in queue, which is also executable.